packages/medusa/src/services/new-totals.ts

```diff
@@ -0,0 +1,707 @@
+import {
+  ITaxCalculationStrategy,
+  TaxCalculationContext,
+  TransactionBaseService,
+} from "../interfaces"
+import { EntityManager } from "typeorm"
+import {
+  Discount,
+  DiscountRuleType,
+  GiftCard,
+  LineItem,
+  LineItemTaxLine,
+  Region,
+  ShippingMethod,
+  ShippingMethodTaxLine,
+} from "../models"
+import { TaxProviderService } from "./index"
+import { LineAllocationsMap } from "../types/totals"
+import TaxInclusivePricingFeatureFlag from "../loaders/feature-flags/tax-inclusive-pricing"
+import { FlagRouter } from "../utils/flag-router"
+import { calculatePriceTaxAmount, isDefined } from "../utils"
+import { MedusaError } from "medusa-core-utils"
+
+type LineItemTotals = {
+  unit_price: number
+  quantity: number
+  subtotal: number
+  tax_total: number
+  total: number
+  original_total: number
+  original_tax_total: number
+  tax_lines: LineItemTaxLine[]
+  discount_total: number
+}
+
+type ShippingMethodTotals = {
+  price: number
+  tax_total: number
+  total: number
+  subtotal: number
+  original_total: number
+  original_tax_total: number
+  tax_lines: ShippingMethodTaxLine[]
+}
+
+type InjectedDependencies = {
+  manager: EntityManager
+  taxProviderService: TaxProviderService
+  taxCalculationStrategy: ITaxCalculationStrategy
+  featureFlagRouter: FlagRouter
+}
+
+export default class NewTotalsService extends TransactionBaseService {
+  protected readonly manager_: EntityManager
+  protected readonly transactionManager_: EntityManager | undefined
+
+  protected readonly taxProviderService_: TaxProviderService
+  protected readonly featureFlagRouter_: FlagRouter
+  protected readonly taxCalculationStrategy_: ITaxCalculationStrategy
+
+  constructor({
+    manager,
+    taxProviderService,
+    featureFlagRouter,
+    taxCalculationStrategy,
+  }: InjectedDependencies) {
+    super(arguments[0])
+
+    this.manager_ = manager
+    this.taxProviderService_ = taxProviderService
+    this.featureFlagRouter_ = featureFlagRouter
+    this.taxCalculationStrategy_ = taxCalculationStrategy
+  }
+
+  /**
+   * Calculate and return the items totals for either the legacy calculation or the new calculation
+   * @param items
+   * @param includeTax
+   * @param calculationContext
+   * @param taxRate
+   * @param useExistingTaxLines Force to use the tax lines of the line item instead of fetching them
+   */
+  async getLineItemsTotals(
```

Comment:
todo: rename so the public API is `getLineItemTotals` (change the protected method to something else).

Reply:
but it takes a collection, it does not make sense to be item totals and not items totals since it is always multiple totals the plural on totals does not represent that. wdyt?